qa: Intermittent failure in wallet_orphanedreward.py

On master (359f72105ba0184fbf998dfd84217c4229dc54ad) Linux x86_64 build (https://bitcoinbuilds.org/index.php?ansilog=d9ab0e6e-e84d-432e-99c9-a7abe23e2413.log#l2645):

<details><summary>log excerpt</summary>
<p>

```
L2644   193/208 - wallet_orphanedreward.py failed, Duration: 4 s
L2645   
L2646   stdout:
L2647   2021-06-07T17:29:05.693000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20210607_131230/wallet_orphanedreward_13
L2648   2021-06-07T17:29:08.981000Z TestFramework (ERROR): Assertion failed
L2649   Traceback (most recent call last):
L2650     File "/home/ubuntu/src/test/functional/test_framework/test_framework.py", line 128, in main
L2651       self.run_test()
L2652     File "/home/ubuntu/src/test/functional/wallet_orphanedreward.py", line 34, in run_test
L2653       assert_equal(self.nodes[1].getbalance(), 10 + 25)
L2654     File "/home/ubuntu/src/test/functional/test_framework/util.py", line 51, in assert_equal
L2655       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
L2656   AssertionError: not(10.00000000 == 35)
L2657   2021-06-07T17:29:09.036000Z TestFramework (INFO): Stopping nodes
L2658   2021-06-07T17:29:09.741000Z TestFramework (WARNING): Not cleaning up dir /tmp/test_runner_₿_🏃_20210607_131230/wallet_orphanedreward_13
L2659   2021-06-07T17:29:09.741000Z TestFramework (ERROR): Test failed. Test logging available at /tmp/test_runner_₿_🏃_20210607_131230/wallet_orphanedreward_13/test_framework.log
L2660   2021-06-07T17:29:09.741000Z TestFramework (ERROR): 
L2661   2021-06-07T17:29:09.741000Z TestFramework (ERROR): Hint: Call /home/ubuntu/src/test/functional/combine_logs.py '/tmp/test_runner_₿_🏃_20210607_131230/wallet_orphanedreward_13' to consolidate all logs
L2662   2021-06-07T17:29:09.741000Z TestFramework (ERROR): 
L2663   2021-06-07T17:29:09.741000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
L2664   2021-06-07T17:29:09.742000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
L2665   2021-06-07T17:29:09.742000Z TestFramework (ERROR): 
```

</p>
</details>

--- a/test/functional/wallet_orphanedreward.py
+++ b/test/functional/wallet_orphanedreward.py
@@ -31,6 +31,7 @@ class OrphanedBlockRewardTest(BitcoinTestFramework):
         # Let the block reward mature and send coins including both
         # the existing balance and the block reward.
         self.nodes[0].generate(150)
+        self.sync_blocks()
         assert_equal(self.nodes[1].getbalance(), 10 + 25)
         txid = self.nodes[1].sendtoaddress(self.nodes[0].getnewaddress(), 30)
 

